axum-core/src/response/mod.rs

```diff
@@ -356,17 +330,10 @@ impl IntoResponse for StatusCode {
     }
 }
 
-impl<H> IntoResponse for H
-where
-    H: IntoResponseHeaders,
-{
+impl IntoResponse for HeaderMap {
```

Comment:
I guess this makes sense. It's really annoying the previous solution worsens rustc's error reporting.

Reply:
Yeah it is 😞 I still think its worth it overall

Reply:
(including the introduction of `IntoResponseHeaders`)

Reply:
Yeah, agreed.